var/spack/repos/builtin/packages/of-iso-advector/package.py

```diff
@@ -0,0 +1,153 @@
+
+    build_script  = 'spack-Allwmake'   # <- Generated by patch() method.
+    build_userdir = 'spack-userdir'    # Build user APPBIN, LIBBIN into here
+    src_dir = 'isoAdvector'
+
+    phases = ['configure', 'build', 'install']
+
+    #
+    # - End of definitions / setup -
+    #
+
+    @property
+    def foam_info(self):
+        """Emit information about openfoam, check for validity"""
+        foam_spec = self.spec['openfoam']
+        if foam_spec.name == 'foam-extend':
+            raise RuntimeError("incompatible with foam-extend")
+        return foam_spec.format('$_$@$%@+$+')
+
+    def patch(self):
+        """Copy additional files or other patching."""
+        assetdir = join_path(self.package_dir, 'assets')
+        for f in self.assets:
+            install(
+                join_path(assetdir, f),
+                join_path(self.stage.source_path, f))
+        # Emit the dependent openfoam version _immediately_
+        # We may have resolved the wrong version, and don't wish to build
+        # it as a dependency!
+        tty.info('Build for {0}'.format(self.foam_info))
+
+    def configure(self, spec, prefix):
+        """Generate spack-config.sh file."""
+        # Local tweaks
+        config = join_path(self.stage.source_path, 'spack-config.sh')
+        with open(config, 'w') as out:
+            out.write(
+                """# Local tweaks for building
+# Handle sub-directories in source
+src_dir='{src_dir}'
+
+# Local build (for user appbin, libbin)
+. ./local-userdir.sh $PWD/{user_dir}
+#
+"""
+                .format(
+                    src_dir=self.src_dir,
+                    user_dir=self.build_userdir))
+
+    def build(self, spec, prefix):
+        """Build using the OpenFOAM Allwmake script, with a wrapper to source
+        its environment first.
+        """
+        args = []
+        if self.parallel:  # Build in parallel? - pass via the environment
+            os.environ['WM_NCOMPPROCS'] = str(make_jobs)
+        builder = Executable(self.build_script)
+        builder(*args)
+
+    def install(self, spec, prefix):
+        """Install under the prefix directory"""
+
+        files = ['README.md']
+        if '+source' in spec:
+            files.extend(['Allwmake', 'Allwclean'])
+
+        for f in files:
+            src = join_path(self.src_dir, f)
+            if os.path.isfile(src):
+                install(src, join_path(self.prefix, f))
+
+        dirs = ['run']
+        if '+source' in spec:
+            dirs.extend(['applications', 'src'])
+
+        for d in dirs:
+            install_tree(
+                join_path(self.src_dir, d),
+                join_path(self.prefix, d))
+
+        # Place directly under 'bin', 'lib'
+        for d in ['bin', 'lib']:
+            install_tree(
+                join_path(self.build_userdir, d),
+                join_path(self.prefix, d))
+
+# -----------------------------------------------------------------------------
```

Comment:
Sure, but I don't have the feeling that this will make the merge happen any quicker.